### PR TITLE
Add origin/altdata to spectrum displays

### LINKED ISSUE #1797: Add ORIGIN to spectrum legend and render altdata as tooltip

### PR DESCRIPTION
**Pull request by kmshin1397**
Display spectrum origin and altdata values in the plot tooltips and the manage data table. Interestingly, it doesn't look like anyone is actually using the origin field for spectra (I didn't see any non-null or non-empty-string values in the fritz production replica database). Maybe seeing it in the tooltip will serve as advertisement so people are aware of it.

Also tweaked the spectrum plot styling to more accurately handle longer lists of spectra.

Origin and altdata columns on the Manage Data page:
spectra

New plot tooltip additions:
Screenshot 2021-06-24 101851

Before of having a large number of spectra (22 here):
Screenshot 2021-06-24 092156
After:
Screenshot 2021-06-24 093143

Closes #1797

### CODE REVIEW

**Comment by guynir42** [comment]
@kmshin1397  I am flying tonight so will be unavailable for reviews for a fews days.

**Comment by kmshin1397** [comment]
> @kmshin1397  I am flying tonight so will be unavailable for reviews for a fews days. 

No problem, I don't think this one is very high priority - just an enhancement

File: static/js/components/ManageDataForm.jsx [hunk 1]
```diff
@@ -384,6 +432,18 @@ const ManageDataForm = ({ route }) => {
         filter: false,
       },
     },
+    {
+      name: "id",
```
Comment (acrellin) [inline static/js/components/ManageDataForm.jsx:436]:
```suggestion
      name: "origin",
```

**Review by acrellin — APPROVED** [review]
One minor comment, otherwise looks good, thanks!

**Comment by kmshin1397** [comment]
@dmitryduev This includes a fix for the issue with the legend being hidden when you have too many spectra. We had another complaint about it, so I thought we should probably should bump this up in priority and get it merged so I'm tagging you :)

**Review by dmitryduev — APPROVED** [review]
LGTM, CI failure unrelated.

**Pull request merged**